src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -0,0 +1,376 @@
+
+public class HaxeSemanticAnnotator implements Annotator {
+  @Override
+  public void annotate(PsiElement element, AnnotationHolder holder) {
+    analyzeSingle(element, holder);
+  }
+
+  static void analyzeSingle(final PsiElement element, AnnotationHolder holder) {
+    if (element instanceof HaxePackageStatement) {
+      PackageChecker.check((HaxePackageStatement)element, holder);
+    }
+    else if (element instanceof HaxeMethod) {
+      MethodChecker.check((HaxeMethod)element, holder);
+    }
+    else if (element instanceof HaxeClass) {
+      ClassChecker.check((HaxeClass)element, holder);
+    }
+  }
+}
+
+class ClassChecker {
+  static public void check(final HaxeClass clazzPsi, final AnnotationHolder holder) {
+    HaxeClassModel clazz = clazzPsi.getModel();
+    checkInterfaces(clazz, holder);
+    checkExtends(clazz, holder);
+    checkInterfacesMethods(clazz, holder);
+  }
+
+  static public void checkExtends(final HaxeClassModel clazz, final AnnotationHolder holder) {
+    HaxeClassReferenceModel reference = clazz.getParentClass();
+    if (reference != null) {
+      HaxeClassModel aClass1 = reference.getHaxeClass();
+      if (aClass1 != null && !aClass1.isClass()) {
+        holder.createErrorAnnotation(reference.getPsi(), "Not a class");
+      }
+    }
+  }
+
+  static public void checkInterfaces(final HaxeClassModel clazz, final AnnotationHolder holder) {
+    for (HaxeClassReferenceModel interfaze : clazz.getImplementingInterfaces()) {
+      if (interfaze.getHaxeClass() == null || !interfaze.getHaxeClass().isInterface()) {
+        holder.createErrorAnnotation(interfaze.getPsi(), "Not an interface");
+      }
+    }
+  }
+
+  static public void checkInterfacesMethods(final HaxeClassModel clazz, final AnnotationHolder holder) {
+    for (HaxeClassReferenceModel reference : clazz.getImplementingInterfaces()) {
+      checkInterfaceMethods(clazz, reference, holder);
+    }
+  }
+
+  static public void checkInterfaceMethods(final HaxeClassModel clazz, final HaxeClassReferenceModel intReference, final AnnotationHolder holder) {
+    final List<HaxeMethodModel> missingMethods = new ArrayList<HaxeMethodModel>();
+    final List<String> missingMethodsNames = new ArrayList<String>();
+
+    if (intReference.getHaxeClass() != null) {
+      for (HaxeMethodModel method : intReference.getHaxeClass().getMethods()) {
+        if (!method.isStatic()) {
+          if (!clazz.hasMethodSelf(method.getName())) {
+            missingMethods.add(method);
+            missingMethodsNames.add(method.getName());
+          }
+        }
+      }
+    }
+
+    if (missingMethods.size() > 0) {
+      Annotation annotation = holder.createErrorAnnotation(intReference.getPsi(), "Not implemented methods: " + StringUtils.join(missingMethodsNames, ", "));
+      annotation.registerFix(new HaxeSemanticIntentionAction("Implement methods") {
+        @Override
+        public void run() {
+          clazz.addMethodsFromPrototype(missingMethods);
```

Comment:
clazz.addMethodsFromPrototype(..)

This is where I think we ought to pop up the UI, rather than simply adding implementations to the class.